cloudfoundry-example/src/main/java/hello/MainVerticle.java

```diff
@@ -56,15 +55,18 @@ public void start(Future<Void> startFuture) throws Exception {
         httpServer.listen(PORT, handler -> {
             if (handler.succeeded()) {
                 logger.info("Listening on port: " + PORT);
+                startFuture.complete();
             } else {
-                logger.error("Failed to bind on port " + PORT + ". Is it being used?");
+                String errorMessage = "Failed to bind on port " + PORT + ". Is it being used?";
+                logger.error(errorMessage);
+                startFuture.fail(errorMessage);
```

Comment:
Actually you'd rather fail the future with the cause `handler.cause()`

Reply:
Ah ok. This is helpful for me to know too